### PR TITLE
Basic verification summary implementation

### PR DESCRIPTION
**Pull request by atomb**
This draft PR implements the necessary plumbing to track everything verified within SAW and report a summary of it all after verification completes. It currently adds one command, `summarize_verification`, which prints a summary of all `Theorem` or `MethodSpec` values created along with various details about each.

This could almost certainly use additional information, but I'm creating a PR now to get feedback from others on what we should include.

One of the key details I want to include is a summary of all assumptions made when verifying JVM methods or LLVM functions. A concise description of all size limitations put in place during verification (as well as any precondition predicates) is probably one of the most important details. Distinguishing between things that have been assumed and those that have been verified is another important one.

### CODE REVIEW

File: src/SAWScript/Crucible/LLVM/X86.hs [hunk 1]
```diff
@@ -782,21 +782,21 @@ checkGoals ::
...
+  TopLevel SolverStats
 checkGoals sym opts sc tactic = do
   gs <- liftIO $ getGoals sym
   liftIO . printOutLn opts Info $ mconcat
     [ "Simulation finished, running solver on "
     , show $ length gs
     , " goals"
     ]
-  forM_ (zip [0..] gs) $ \(n, g) -> do
+  stats <-
```
Comment (andreistefanescu) [inline src/SAWScript/Crucible/LLVM/X86.hs:793]: nit: keep this on one line

File: src/SAWScript/Interpreter.hs [hunk 2]
```diff
@@ -2392,6 +2393,13 @@ primitives = Map.fromList
     (pureVal disable_crucible_profiling)
     Current
     ["Stop recording profiling information."]
+
+  , prim "summarize_verification" "TopLevel ()"
+    (pureVal summarize_verification)
+    Experimental
+    [ "Provide a human-readable summary of all verifications performed"
```
Comment (andreistefanescu) [inline src/SAWScript/Interpreter.hs:2400]: nit: "Print a human-readable summary of all verification tasks performed."

File: src/SAWScript/VerificationSummary.hs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      item txt = "*" <+> txt
+      code txt = vsep ["~~~~", txt, "~~~~"]
+      subitem = indent 4 . item
+      sectionWithItems _ _ [] = mempty
+      sectionWithItems nm prt items =
+        vsep [section nm, "", vsep (map prt items)]
+      verifStatus s = if Set.null (solverStatsSolvers (s ^. CMS.csSolverStats))
+                      then "assumed"
+                      else "verified"
+      condStatus s = (if null (s ^. (CMS.csPreState . CMS.csConditions))
```
[the remark this answers is not in the document]
Comment (atomb) [reply src/SAWScript/VerificationSummary.hs:69]: Yeah, ultimately I want to include all of those things, but you're probably right that it's misleading until they're included, too.

File: src/SAWScript/VerificationSummary.hs [hunk 4]
```diff
@@ -0,0 +1,96 @@
+{-# LANGUAGE OverloadedStrings #-}
+{- |
+Module      : SAWScript.VerificationSummary
+Description : Summaries of verification for human consumption.
+License     : BSD3
+Maintainer  : atomb
+-}
+
+module SAWScript.VerificationSummary where
```
[the remark this answers is not in the document]
Comment (atomb) [reply src/SAWScript/VerificationSummary.hs:9]: Good catch.

File: src/SAWScript/Interpreter.hs (hunk 2 above)
Comment (atomb) [reply src/SAWScript/Interpreter.hs:2400]: Fixed.

File: src/SAWScript/Crucible/LLVM/X86.hs (hunk 1 above)
Comment (atomb) [reply src/SAWScript/Crucible/LLVM/X86.hs:793]: Fixed.

**Pull request merged**